packages/openid4vc/src/openid4vc-issuer/OpenId4VcIssuerApi.ts

```diff
@@ -72,33 +67,39 @@ export class OpenId4VcIssuerApi {
    */
   public async createCredentialOffer(options: OpenId4VciCreateCredentialOfferOptions & { issuerId: string }) {
     const { issuerId, ...rest } = options
-    const issuer = await this.openId4VcIssuerService.getByIssuerId(this.agentContext, issuerId)
+    const issuer = await this.openId4VcIssuerService.getIssuerByIssuerId(this.agentContext, issuerId)
     return await this.openId4VcIssuerService.createCredentialOffer(this.agentContext, { ...rest, issuer })
   }
 
   /**
-   * This function retrieves the credential offer referenced by the given URI.
-   * Retrieving a credential offer from a URI is possible after a credential offer was created with
-   * @see createCredentialOffer and the credentialOfferUri option.
-   *
-   * @throws if no credential offer can found for the given URI.
-   * @param uri - The URI referencing the credential offer.
-   * @returns The credential offer payload associated with the given URI.
+   * This function creates a response which can be send to the holder after receiving a credential issuance request.
    */
-  public async getCredentialOfferFromUri(uri: string): Promise<OpenId4VciCredentialOfferPayload> {
-    return await this.openId4VcIssuerService.getCredentialOfferFromUri(this.agentContext, uri)
+  public async createCredentialResponse(
+    options: OpenId4VciCreateCredentialResponseOptions & { issuanceSessionId: string }
+  ) {
+    const { issuanceSessionId, ...rest } = options
+    const issuanceSession = await this.openId4VcIssuerService.getIssuanceSessionById(
```

Comment:
Method naming seems inconsistent (see previous comment)

Reply:
This is for getting an issuance session by the record id (not the issuerId), so it's consistent, just not very logical :(

Reply:
Ah, makes sense